crates/core/tedge_api/src/workflow/state.rs

```diff
@@ -273,6 +274,11 @@ impl GenericCommandState {
         GenericCommandState::extract_text_property(&self.payload, REASON)
     }
 
+    /// Return value of the result field
+    pub fn result(&self) -> &Value {
+        self.payload.get(RESULT).unwrap_or(&Value::Null)
+    }
```

Comment:
This sounds a bit too specific, this result being only used by C8y custom operations.

The payload being `pub`, this one-liner can be done directly in `c8y_mapper_ext`.

That said, what is proposed by Albin is by far a better approach. See https://github.com/thin-edge/thin-edge.io/pull/3283#issuecomment-2535137925